007GameManagerLibrary/_007GameManager.cs

```diff
@@ -105,27 +105,32 @@ public void Leave(string name, bool died)
             if (callbacks.ContainsKey(name))
             {
                 callbacks.Remove(name);
-                NotifyPlayers();
 
-                //Reset the round as someone might have selected them as a target
-                playerRounds.Clear();
+                NotifyPlayers();
 
-                if (!died && callbacks.Count > 1)
-                    foreach (var cb in callbacks.Values)
-                        cb.ResetRound();
-                else if (callbacks.Count == 1)
+                //Only reset the round if the round isn't in progress
+                if (gameInProgress)
                 {
-                    PlayerRound round = new PlayerRound("", PlayerActions.Shoot)
+                    //Reset the round as someone might have selected them as a target
+                    playerRounds.Clear();
+
+                    if (!died && callbacks.Count > 1)
+                        foreach (var cb in callbacks.Values)
+                            cb.ResetRound();
+                    else if (callbacks.Count == 1)
                     {
-                        Results = new List<string>() { "You are the last player standing!" },
-                        HealthLost = 0
-                    };
-                    callbacks.First().Value.SendRoundResults(round);
-                } 
-                else if (callbacks.Count == 0)
-                    gameInProgress = false;
-                Console.WriteLine($"Players remaining: {callbacks.Count}");
-            }  
+                        PlayerRound round = new PlayerRound("", PlayerActions.Shoot)
+                        {
+                            Results = new List<string>() { "You are the last player standing!" },
+                            HealthLost = 0
+                        };
+                        callbacks.First().Value.SendRoundResults(round);
```

Comment:
on second thought I guess we should just have a second constructor that just takes a single message to send, my bad